src/k8s/api/v1/types.go

```diff
@@ -69,94 +98,72 @@ func (c ClusterStatus) HaClusterFormed() bool {
-		result.WriteString("  standby-nodes:\n")
-		for _, standBy := range standBys {
-			result.WriteString(fmt.Sprintf("    - %s\n", standBy.Address))
-		}
-	} else {
-		result.WriteString("  standby-nodes: none\n")
-	}
-	if len(spares) > 0 {
-		result.WriteString("  spare-nodes:\n")
-		for _, spare := range spares {
-			result.WriteString(fmt.Sprintf("    - %s\n", spare.Address))
-		}
-	} else {
-		result.WriteString("  spare-nodes: none\n")
-	}
-
-	return result.String()
-}
 
 // TODO: Print k8s version. However, multiple nodes can run different version, so we would need to query all nodes.
 func (c ClusterStatus) String() string {
 	result := strings.Builder{}
 
+	// longer than the longest key (eye-balled), make the output left aligned
+	maxLen := 25
+
 	// Status
 	if c.Ready {
-		result.WriteString("status: ready")
+		result.WriteString(fmt.Sprintf("%-*s %s", maxLen, "cluster status:", "ready"))
+	} else {
+		result.WriteString(fmt.Sprintf("%-*s %s", maxLen, "cluster status:", "not ready"))
+	}
+	result.WriteString("\n")
+
+	// Control Plane Nodes
+	result.WriteString(fmt.Sprintf("%-*s ", maxLen, "control plane nodes:"))
+	if len(c.Members) > 0 {
+		members := make([]string, 0, len(c.Members))
+		for _, m := range c.Members {
+			if m.ClusterRole == ClusterRoleControlPlane {
+				members = append(members, fmt.Sprintf("%s (%s)", m.Address, m.DatastoreRole))
+			}
+		}
+		result.WriteString(strings.Join(members, ", "))
 	} else {
-		result.WriteString("status: not ready")
+		result.WriteString("none")
 	}
 	result.WriteString("\n")
 
 	// High availability
-	result.WriteString("high-availability: ")
+	result.WriteString(fmt.Sprintf("%-*s ", maxLen, "high availability:"))
 	if c.HaClusterFormed() {
 		result.WriteString("yes")
 	} else {
 		result.WriteString("no")
 	}
-
-	// Datastore
 	result.WriteString("\n")
-	result.WriteString("datastore:\n")
-	result.WriteString(c.datastoreToString())
 
-	// Config
-	if !c.Config.Empty() {
-		b, _ := yaml.Marshal(c.Config)
-		result.WriteString(string(b))
+	// Datastore
+	// TODO: how to understand if the ds is running or not?
+	if c.Datastore.Type != "" {
+		result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "datastore:", c.Datastore.Type))
+	} else {
+		result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "datastore:", "disabled"))
 	}
+
+	// Network
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "network:", c.Network.GetMessage()))
+
+	// DNS
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "dns:", c.DNS.GetMessage()))
+
+	// Ingress
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "ingress:", c.Ingress.GetMessage()))
+
+	// Load Balancer
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "load-balancer:", c.LoadBalancer.GetMessage()))
+
+	// Local Storage
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "local-storage:", c.LocalStorage.GetMessage()))
+
+	// Gateway
+	result.WriteString(fmt.Sprintf("%-*s %s\n", maxLen, "gateway", c.Gateway.GetMessage()))
+
```

Comment:
this can just be 6 lines, again i think it's OK to repeat 25 a few times:

```
result.WriteString(fmt.Sprintf("%25s %s", "network:", c.Network.GetMessage()))
result.WriteString(fmt.Sprintf("%25s %s", "dns:", c.DNS.GetMessage()))
....
```

Also, you can change `GetMessage() string` to a `String()`, then use `fmt.Sprintf("%v", c.Network))`

Reply:
^ most importantly, the comments themselves do not add much here, it's ok to remove them